go-controller/pkg/ovn/egressgw.go

```diff
@@ -390,42 +388,45 @@ func (oc *Controller) deletePodGWRoutesForNamespace(pod, namespace string) {
 		// check for previously configured pod routes
 		routeInfos := oc.getRouteInfosForGateway(gwIP.String(), namespace)
 		for _, routeInfo := range routeInfos {
-			for podIP, route := range routeInfo.podExternalRoutes {
-				for routeGwIP, gr := range route {
-					if gwIP.String() != routeGwIP {
-						continue
-					}
-					if gr == "" {
-						continue
-					}
-					mask := GetIPFullMask(podIP)
-					node := util.GetWorkerFromGatewayRouter(gr)
-					portPrefix, err := oc.extSwitchPrefix(node)
-					if err != nil {
-						klog.Infof("Failed to find ext switch prefix for %s %v", node, err)
-						continue
-					}
+			routeInfo.Lock()
+			if !routeInfo.deleted {
```

Comment:
how about if routeInfo.deleted { continue } ? save indent on the nested loops

Reply:
If you're cool with:

```
routeInfo.Lock()
if routeInfo.deleted {
    routeInfo.Unlock()
    continue
}
...
routeInfo.Unlock()
```